snapcraft/plugins/ant.py

```diff
@@ -59,22 +102,88 @@ def schema(cls):
             "items": {"type": "string"},
             "default": [],
         }
+
+        schema["properties"]["ant-version"] = {"type": "string"}
+
+        schema["properties"]["ant-version-checksum"] = {"type": "string"}
+
+        schema["required"] = ["source"]
+        schema["properties"]["ant-openjdk-version"] = {"type": "string", "default": ""}
+
         schema["required"] = ["source"]
         return schema
 
-    def __init__(self, name, options, project):
-        super().__init__(name, options, project)
-        self.build_packages.append("ant")
+    @classmethod
+    def get_pull_properties(cls):
+        # Inform Snapcraft of the properties associated with pulling. If these
+        # change in the YAML Snapcraft will consider the pull step dirty.
+        return ["ant-version", "ant-version-checksum", "ant-openjdk-version"]
 
     @classmethod
     def get_build_properties(cls):
         # Inform Snapcraft of the properties associated with building. If these
         # change in the YAML Snapcraft will consider the build step dirty.
         return ["ant-build-targets", "ant-properties"]
 
+    @property
+    def _ant_tar(self):
```

Comment:
What is the benefit of doing this lazily instead of just setting it in `_setup_ant`?

Reply:
Legacy today most likely, when project_loader was called regardless, so if on a weird system we would fail as all the plugin's __init__'s were called. I say we keep the safe measures to not load the module until needed.

Reply:
Haha, how can this be legacy? You just added it :laughing: . The risk here is that future devs may try to use `self._ant_tar` from within `__init__` before `setup_ant()` is called, which will blow up.

Reply:
ok, but this goes into the TODO bucket, as this is the same pattern in:
- nodejs
- maven
- ant
- gradle